### PR TITLE
Feat/integ 688

### PR DESCRIPTION
**Pull request by rusticpenguin**
## Purpose

We want the user to be able to automagically be able to rewrite their article's tone, length, etc

## Approach

Created a new feature in AICG to host this functionality.

- moved the generator logic to the top level of a generator component
- added a rewrite generator component
- allowed for an infinite amount of args to be passed into the prompts
- created a paired button/input field
- updated to vite V4 so we can have better tests

### CODE REVIEW

File: apps/ai-content-generator/src/locations/Dialog.tsx [hunk 1]
```diff
@@ -26,14 +27,24 @@ const Dialog = () => {
...
 
+  const getGenerator = (feature: AIFeature) => {
+    switch (feature) {
+      case AIFeature.REWRITE:
+        return <RewriteGenerator />;
+
+      default:
+        return <CommonGenerator />;
+    }
+  };
```
Comment (rusticpenguin) [inline apps/ai-content-generator/src/locations/Dialog.tsx:38]: it's back 😦

**Review by whitelisab — COMMENTED** [review]
Overall the code looks really great! 💪 I mostly had comments around matching the design and some questions about the behavior of the tone option buttons.

File: apps/ai-content-generator/src/components/app/dialog/rewrite-generator/button-text-field/ButtonTextField.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            css={styles.button}
+            key={buttonLabel}
+            onClick={() => handleButtonClick(buttonLabel)}
+            isActive={activeButtons.has(buttonLabel.toLowerCase())}>
+            {buttonLabel}
+          </Button>
+        ))}
+      </Flex>
+
+      <FormLabel>Additional instructions</FormLabel>
```
[the remark this answers is not in the document]
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/app/dialog/rewrite-generator/button-text-field/ButtonTextField.tsx:103]: I'll sync with Emily on both this and the buttons! I feel this shouldn't say optional

File: apps/ai-content-generator/src/configs/prompts/rewritePrompt.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const rewritePrompt = (
+  prompt: string,
+  targetLocale: string,
+  rewriteActions: string
+) => `You are editing content generated by someone else, and need to rewrite the following text using the following constraints/tonal shifts: "${rewriteActions}". Keep in mind that the constraints can be categorized in the following ways: tone changes, adjustments, edits, emulating certain writing styles, and so on.  Use your best judgement in editing the article given the list of constraints. Your job is to just rewrite the content and not expand it, keeping the word count of the generated text similar to the word count of the original content, unless if the constraints listed includes longer. You cannot produce content that is longer in word count than the content quoted below unless if the constraints listed includes longer
```
[the remark this answers is not in the document]
Comment (rusticpenguin) [reply apps/ai-content-generator/src/configs/prompts/rewritePrompt.ts:5]: The dreaded side comments by GPT.

I think I have a good way of fixing that with system messages, I just need to fully test it

File: apps/ai-content-generator/src/components/app/dialog/rewrite-generator/button-text-field/ButtonTextField.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import { useEffect, useState } from 'react';
+import { useDebounce } from 'usehooks-ts';
+import { styles } from './ButtonTextField.styles';
+
+interface Props {
+  inputValue: string;
+  handleInputChange: (inputData: string | ((prev: string) => string)) => void;
+}
+
+export const buttons = ['Shorter', 'Longer', 'Casual', 'Formal', 'Simple', 'Witty', 'Tolkien'];
```
[the remark this answers is not in the document]
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/app/dialog/rewrite-generator/button-text-field/ButtonTextField.tsx:11]: I think a toggling feature could be cool! let's float this to the team for potential improvements

**Review by whitelisab — APPROVED** [review]
Approving this, as I think the remaining comments might warrant some discussion and can be addressed in follow up work!

**Pull request merged**